SU2_CFD/include/numerics/turbulent/turb_diffusion.hpp

```diff
@@ -70,21 +99,64 @@ class CAvgGrad_TurbSA final : public CAvgGrad_Scalar {
  * \ingroup ViscDiscr
  * \author F. Palacios
  */
-class CAvgGrad_TurbSA_Neg final : public CAvgGrad_Scalar {
+template <class FlowIndices>
+class CAvgGrad_TurbSA_Neg final : public CAvgGrad_Scalar<FlowIndices> {
 private:
+  using Base = CAvgGrad_Scalar<FlowIndices>;
+  using Base::Laminar_Viscosity_i;
+  using Base::Laminar_Viscosity_j;
+  using Base::Density_i;
+  using Base::Density_j;
+  using Base::ScalarVar_i;
+  using Base::ScalarVar_j;
+  using Base::Proj_Mean_GradScalarVar;
+  using Base::proj_vector_ij;
+  using Base::implicit;
+  using Base::Flux;
+  using Base::Jacobian_i;
+  using Base::Jacobian_j;
+
   const su2double sigma = 2.0/3.0;
   const su2double cn1 = 16.0;
 
   /*!
    * \brief Adds any extra variables to AD
    */
-  void ExtraADPreaccIn(void) override;
+  void ExtraADPreaccIn() override {}
 
   /*!
-   * \brief SA specific steps in the ComputeResidual method
+   * \brief SA-neg specific steps in the ComputeResidual method
    * \param[in] config - Definition of the particular problem.
    */
-  void FinishResidualCalc(const CConfig* config) override;
+  void FinishResidualCalc(const CConfig* config) override {
+    /*--- Compute mean effective viscosity ---*/
+
+    su2double nu_i = Laminar_Viscosity_i/Density_i;
+    su2double nu_j = Laminar_Viscosity_j/Density_j;
+
+    su2double nu_ij = 0.5*(nu_i+nu_j);
+    su2double nu_tilde_ij = 0.5*(ScalarVar_i[0] + ScalarVar_j[0]);
+
+    su2double nu_e;
+
+    if (nu_tilde_ij > 0.0) {
+      nu_e = nu_ij + nu_tilde_ij;
+    }
+    else {
+      su2double Xi = nu_tilde_ij/nu_ij;
+      su2double fn = (cn1 + Xi*Xi*Xi)/(cn1 - Xi*Xi*Xi);
+      nu_e = nu_ij + fn*nu_tilde_ij;
+    }
+
+    Flux[0] = nu_e*Proj_Mean_GradScalarVar[0]/sigma;
```

Comment:
There was a bug here for SA-neg as it was using the "uncorrected flux" which is prone to odd-even decoupling.
To keep track of this I've changed the scope of the PR.

The intrusive part of the changes (for @TobiKattmann and @suargi ) is done, so if it is ok with you I propose merging this now, to use it also in the species solver and refactored SA sources.

@WallyMaier please continue the NEMO turbulence in another PR, it should work out of the box now.

Reply:
I'll give this PR a read later today 👍